frontend/lib/screens/new_student/local_widget/student_form.dart

```diff
@@ -0,0 +1,209 @@
+                      if (_formKey.currentState.validate()) {
+                        widget.changeStep(false);
+                        
+                      }
+                    },
+                    child: Text(
+                      'Continue',
+                      style: TextStyle(color: Colors.white),
+                    ),
+                  ),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _createnameField(Student student) {
+    return TextFormField(
+      decoration: const InputDecoration(
+        hintText: 'Name',
+      ),
+      initialValue: student.name ?? '',
+      onChanged: (value) {
+        setState(() {
+          student.name = value;
+        });
+      },
+      validator: (value) {
+        if (value.isEmpty) {
+          return 'Please enter a student name';
+        }
+        return null;
+      },
+    );
+  }
+
+Widget _createsurnameField(Student student) {
+    return TextFormField(
+      decoration: const InputDecoration(
+        hintText: 'Surname',
+      ),
+      initialValue: student.surname ?? '',
+      onChanged: (value) {
+        setState(() {
+          student.surname = value;
+        });
+      },
+      validator: (value) {
+        if (value.isEmpty) {
+          return 'Please enter a student surname';
+        }
+        return null;
+      },
+    );
+  }
+
+  Widget _createemailField(Student student) {
+    return TextFormField(
+      decoration: const InputDecoration(
+        hintText: 'student@school.com',
+      ),
+      initialValue: student.email ?? '',
+      onChanged: (value) {
+        setState(() {
+          student.email = value;
+        });
+      },
+      validator: (value) {
+        if (value.isEmpty) {
+          return 'Please enter a student email';
+        }
+        return null;
+      },
+    );
+  }
+
+
+
+  Widget _createDescriptionField(Student student) {
+    return TextFormField(
+      decoration: const InputDecoration(
+        hintText: "Try to be as descriptive as possible",
+        labelText: "Tell about yourself",
+        filled: true,
+      ),
+      initialValue: student.description ?? '',
+      onChanged: (value) {
+        setState(() {
+          student.description = value;
+        });
+      },
+      maxLines: 4,
+      validator: (value) {
+        if (value.isEmpty) {
+          return 'Please enter some text';
+        }
+        return null;
+      },
+    );
+  }
+
+  Widget _createphoneField(Student student) {
+    return TextFormField(
+      decoration: const InputDecoration(
+        hintText: 'Phone number',
+      ),
+      initialValue: student.phone ?? '',
+      onChanged: (value) {
+        setState(() {
+          student.phone = value;
+        });
+      },
+      validator: (value) {
+        if (value.isEmpty) {
+          return 'Please enter a phone number';
+        }
+        return null;
+      },
+    );
+  }
```

Comment:
@BassamZ it would be great if this field would take only numbers as an input. You can check an example in the creation of the placement, with the salary or the hours per week field